tls/s2n_config.c

```diff
@@ -308,32 +323,51 @@ int s2n_config_set_client_auth_type(struct s2n_config *config, s2n_cert_auth_typ
     return 0;
 }
 
-int s2n_config_set_verify_cert_chain_cb(struct s2n_config *config, verify_cert_trust_chain_fn *callback, void *context)
+int s2n_config_set_ct_support_level(struct s2n_config *config, s2n_ct_support_level type)
 {
     notnull_check(config);
-    notnull_check(callback);
-
-    config->verify_cert_chain_cb = callback;
-    config->verify_cert_context = context;
+    config->ct_type = type;
 
     return 0;
 }
 
-int s2n_config_set_ct_support_level(struct s2n_config *config, s2n_ct_support_level type)
-{
-    config->ct_type = type;
+int s2n_config_set_verify_host_callback(struct s2n_config *config, s2n_verify_host_fn verify_host_fn, void *data) {
+    notnull_check(config);
+    config->verify_host = verify_host_fn;
+    config->data_for_verify_host = data;
+    return 0;
+}
 
+int s2n_config_set_check_stapled_ocsp_response(struct s2n_config *config, uint8_t check_ocsp) {
+    notnull_check(config);
+    config->check_ocsp = check_ocsp;
     return 0;
 }
 
 
 int s2n_config_set_status_request_type(struct s2n_config *config, s2n_status_request_type type)
 {
+    if(type == S2N_STATUS_REQUEST_OCSP && !s2n_x509_ocsp_stapling_supported()) {
+        return -1;
+    }
+
+    notnull_check(config);
     config->status_request_type = type;
 
     return 0;
 }
 
+int s2n_config_set_verification_ca_location(struct s2n_config *config, const char *ca_file_pem, const char *ca_dir) {
+    notnull_check(config);
+    int err_code = s2n_x509_trust_store_from_ca_file(&config->trust_store, ca_file_pem, ca_dir);
+
+    if(!err_code) {
+        config->status_request_type = s2n_x509_ocsp_stapling_supported() ? S2N_STATUS_REQUEST_OCSP : S2N_STATUS_REQUEST_NONE;
```

Comment:
sort of, but that's an explicit action from the user instead of us making a sane default decision.

set_status_request_type returns an error if ocsp stapling isn't supported for the client. In setup of the trust store, we want to setup the verification side of things as secure as can be supported. So the APIs are a different use case IMHO.